### PR TITLE
Read(lookup) shards by key

### PR DESCRIPTION
**Pull request by honnix**
To get strongly consistent shard usage.

### CODE REVIEW

File: styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java [hunk 1]
```diff
@@ -732,6 +735,18 @@ public void runInTransactionShouldThrowIfDatastoreNewTransactionFails() throws E
     verify(transactionFunction, never()).apply(any());
   }
 
+  @Test
+  public void shouldReturnShardsForCounter() throws Exception {
+    storage.runInTransaction(tx -> {
+      tx.store(Shard.create(COUNTER_ID1, 1, 3));
```
Comment (honnix) [inline styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: Let's at least create two? ;)

**Comment by honnix** [comment]
👍

**Comment by ulzha** [comment]
Note it's not guaranteed to be a strongly consistent snapshot of the entire list of shards: AFAIU each lookup by key in the batch `get` is still an independent lookup unless they're placed in a transaction.

Anyway this is necessary for avoiding the unpredictably delayed consistency of global queries...

File: styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java [hunk 2]
```diff
@@ -734,6 +735,19 @@ public void runInTransactionShouldThrowIfDatastoreNewTransactionFails() throws E
...
 
+  @Test
+  public void shouldReturnShardsForCounter() throws Exception {
+    storage.runInTransaction(tx -> {
+      tx.store(Shard.create(COUNTER_ID1, 0, 0));
+      tx.store(Shard.create(COUNTER_ID1, 1, 3));
+      return null;
+    });
+    final Map<Integer, Long> map = storage.shardsForCounter(COUNTER_ID1);
+    assertEquals(2, map.size());
```
Comment (ulzha) [inline styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: Minor nitpick: all the other tests use `assertThat` and `is`, for consistent and good readability of messages.

Comment (honnix) [reply styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: That has been a question mark in my head for long time. What is the benefit of using assertThat?

**Pull request merged**